channeldb/graph.go

```diff
@@ -332,51 +443,29 @@ func (c *ChannelGraph) NewPathFindTx() (kvdb.RTx, error) {
 func (c *ChannelGraph) ForEachChannel(cb func(*ChannelEdgeInfo,
 	*ChannelEdgePolicy, *ChannelEdgePolicy) error) error {
 
-	// TODO(roasbeef): ptr map to reduce # of allocs? no duplicates
-
-	return kvdb.View(c.db, func(tx kvdb.RTx) error {
-		// First, grab the node bucket. This will be used to populate
-		// the Node pointers in each edge read from disk.
-		nodes := tx.ReadBucket(nodeBucket)
-		if nodes == nil {
+	return c.db.View(func(tx kvdb.RTx) error {
+		edges := tx.ReadBucket(edgeBucket)
+		if edges == nil {
```

Comment:
This changes the logic of when `ErrGraphNotFound` vs. `ErrGraphNoEdgesFound` is returned. I think `ErrGraphNotFound` should only be returned if the `nodeBucket` is not present. We do check for `ErrGraphNoEdgesFound` in some call sites, so we should be careful with when exactly we return what.

Reply:
Changed to `ErrGraphNoEdgesFound`